### ISSUE TITLE
[Zone resilient VM] [CLI] Enable new capability for new and existing VM

### ISSUE THREAD

**Issue by Aarthi-Vijaya-raghavan** [Azure/azure-cli#29807]
### Preconditions

- [X] No need to upgrade Python SDK or the Python SDK is ready.

### Resource Provider

Microsoft.Compute/Virtual Machines

### Description of Feature or Work Requested

ASK: 
Zone resilient VM will be a capability which customers can enable as part of new VM create and existing VM. The enablement of the feature will be post the validations are successful. Once enabled there would be a move option that would show up next to the Availability zone which will allow customers to move their VMs across AZ during any kind of outage.
Ask is to combine 3 API in order when customer provides a PATCH VM with new/different AZ. 
Force deallocate API
PUT VM API
Start VM API
Strategy doc - [Zone-Resilient-VMs-Strategy.docx](https://microsoftapc.sharepoint.com/:w:/t/AzureCoreIDC/EVEZRobaCMBHvhaMrb9wIWUB0rKa34zIOK6xSHcju_HE7Q?e=kMt6B0)
PM Spec - [Zone-Resilient-VMs-ZR-components-PM-Spec.docx](https://microsoftapc.sharepoint.com/:w:/t/AzureCoreIDC/ES0F2aTy9M5Pt-xLQOaeDnYB7FopaAJnfptHWO1PySGtWA?e=nyksmO)
Design doc - [Reliable VM.docx](https://microsoft.sharepoint.com/:w:/t/AzureIDC/AzureIDC_CRP/EeRboy_bCtNAgoqZYtJ46LABjxaKGQ1poyY_kCjLAdWyhw?e=tcRWU6)

API will be ready by Oct 2024. 

### Minimum API Version Required

2025-04-01

### Swagger PR link / SDK link

https://github.com/Azure/azure-rest-api-specs/pull/39385
https://github.com/Azure/azure-rest-api-specs/pull/39377

### Target Date

2026-03-01

### PM Contact

aarthiv@microsoft.com

### Engineer Contact

kirankumarp@microsoft.com

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by zhoxing-ms** [comment]
@Aarthi-Vijaya-raghavan May I ask if this feature request is related to security? Please note that currently, due to the CLI team being busy with the security wave and prioritizing security in this semester, so all the feature requests unrelated to security will be postponed. cc PM @Jacekey23

**Comment by Aarthi-Vijaya-raghavan** [comment]
@zhoxing-ms this feature is not related to security but related to resiliency. This is a new feature we are working to improve resiliency of VMs by allowing them to be moved across zones during a zonal outage.

**Comment by Aarthi-Vijaya-raghavan** [comment]
@zhoxing-ms filing this for Br semester. We are targeting this feature for public preview by Jul'25 and would need CLI support. The API side are being implemented and is expected to be ready by May'25. Please let me know how much time is required for CLI team to enable this feature once API is ready? We are looking for additional efforts to encompass the multiple APIs to a single command on CLI.

**Comment by zhoxing-ms** [comment]
Same as https://github.com/Azure/azure-cli/issues/29808#issuecomment-2661913935

**Comment by Aarthi-Vijaya-raghavan** [comment]
@zhoxing-ms  Can CLI team pick this up in Jan 2026 or Feb 2026. We are working towards public preview for this feature in Mar 2026 and need CLI support as P0.

**Comment by william051200** [comment]
While reviewing the setup, I noticed that two branches have not yet been merged, and our development depends heavily on these updates:

main ← [cplat-2025-11-01](https://github.com/Azure/azure-rest-api-specs/pull/39476) ← [bhavikatambi/resiliencyProfileChanges](https://github.com/Azure/azure-rest-api-specs/pull/39377) (pending)
main ← [cplat-2025-11-01](https://github.com/Azure/azure-rest-api-specs/pull/39476) ← [bhavikatambi/forceDeallocate](https://github.com/Azure/azure-rest-api-specs/pull/39377) (pending)

To reduce risk and ensure stable testing, we plan to proceed with development after these changes are merged into main.

cc @yanzhudd

**Comment by yanzhudd** [comment]
Hi @Aarthi-Vijaya-raghavan 
please let us know if the spec definition has been published to the main branch
